From c629f91a3bec7b33def2a52a90a9a4fca1304520 Mon Sep 17 00:00:00 2001
From: Zhao Yakui <yakui.zhao@intel.com>
Date: Fri, 26 Sep 2014 09:18:24 +0800
Subject: [PATCH 6/8] [Libva] [PATCH V3 2/3] Use the strncasecmp instead of
 strcasecmp to make it more robust

At the same time the extra string arrary is removed.

Signed-off-by: Zhao Yakui <yakui.zhao@intel.com>
---
 src/i965_device_info.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/src/i965_device_info.c b/src/i965_device_info.c
index a7ea66c..f21f52d 100644
--- a/src/i965_device_info.c
+++ b/src/i965_device_info.c
@@ -415,13 +415,12 @@ const static char *hsw_cpu_hook_list[] =  {
 
 static void hsw_hw_codec_preinit(VADriverContextP ctx, struct hw_codec_info *codec_info)
 {
-    char model_string[64], model_id[64];
+    char model_string[64];
     char *model_ptr, *tmp_ptr;
     int i, model_len, list_len;
     bool found;
 
     memset(model_string, 0, sizeof(model_string));
-    memset(model_id, 0, sizeof(model_id));
 
     /* If it can't detect cpu model_string, leave it alone */
     if (intel_driver_detect_cpustring(model_string))
@@ -430,13 +429,10 @@ static void hsw_hw_codec_preinit(VADriverContextP ctx, struct hw_codec_info *cod
     /* strip the cpufreq info */
     model_ptr = model_string;
     tmp_ptr = strstr(model_ptr, "@");
-    if (!tmp_ptr)
-	    return;
-
     *tmp_ptr = '\0';
 
     /* strip the space character and convert to the lower case */
-    model_ptr = model_id;
+    model_ptr = model_string;
     model_len = strlen(model_string);
     for (i = 0; i < model_len; i++) {
          if (model_string[i] != ' ') {
@@ -448,14 +444,14 @@ static void hsw_hw_codec_preinit(VADriverContextP ctx, struct hw_codec_info *cod
 
     found = false;
     list_len = sizeof(hsw_cpu_hook_list) / sizeof(char *);
-    model_len = strlen(model_id);
+    model_len = strlen(model_string);
     for (i = 0; i < list_len; i++) {
         model_ptr = (char *)hsw_cpu_hook_list[i];
 
         if (strlen(model_ptr) != model_len)
 	    continue;
 
-        if (strncasecmp(model_id, model_ptr, model_len) == 0) {
+        if (strncasecmp(model_string, model_ptr, model_len) == 0) {
             found = true;
             break;
 	}
-- 
2.1.0.rc2.206.gedb03e5

